modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/odata/ExpressionVisitorImpl.java

```diff
@@ -338,6 +350,58 @@ private String _getPropertyName(Object object) {
 		return String.valueOf(object);
 	}
 
+	private Object _getSub(Object left, Object right) {
+		Duration duration = Duration.parse(right.toString());
+
+		if (MethodExpression.Type.NOW.equals(left)) {
+			long days = duration.toDays();
+
+			if (days <= 1) {
+				return "last24Hours";
+			}
+			else if (days <= 2) {
+				return "yesterday";
+			}
+			else if (days <= 7) {
+				return "last7Days";
+			}
+			else if (days <= 28) {
+				return "last28Days";
+			}
+			else if (days <= 30) {
+				return "last30Days";
+			}
+			else {
+				return "last90Days";
+			}
+		}
+
+		throw new UnsupportedOperationException(
+			StringBundler.concat(
+				"Unsupported types in _getSubOperationObject with  Arithmetic ",
+				"Operator SUB with left type", left.getClass(),
+				"and right type", right.getClass()));
+	}
+
+	private Object _normalizeDurationLiteral(String literal)
+		throws ExpressionVisitException {
+
+		literal = StringUtil.removeSubstring(literal, "duration");
+
+		literal = StringUtil.unquote(literal);
+
+		try {
+			Duration.parse(literal);
+		}
+		catch (DateTimeParseException dateTimeParseException) {
+			throw new ExpressionVisitException(
+				"Invalid duration, use a correct format:" +
+					dateTimeParseException.getMessage());
+		}
+
+		return literal;
```

Comment:
What do you think about returning the duration?

So you have the right type.

Reply:
Yes, well seen.